### PR TITLE
Merkleblock Message

### PR DESCRIPTION
**Pull request by throughnothing**

### CODE REVIEW

**Comment by throughnothing** [comment]
I believe I just need to add tests for this once the relevant bitcore changes are in here: https://github.com/bitpay/bitcore/pull/1102

**Comment by maraoz** [comment]
Excellent. Will review the other required bitcore PR first
closes https://github.com/bitpay/bitcore-p2p/issues/10

File: package.json [hunk 1]
```diff
@@ -52,7 +52,7 @@
     "url": "https://github.com/bitpay/bitcore-p2p.git"
   },
   "dependencies": {
-    "bitcore": "^0.10.2",
+    "bitcore": "^0.11.0",
```
Comment (throughnothing) [inline package.json]: Wasn't sure if I should include this in the PR or not.

File: lib/messages.js [hunk 2]
```diff
@@ -941,5 +973,7 @@ Buffers.prototype.skip = function(i) {
 
 [Inventory, GetData, NotFound].forEach(function(clazz) {
   clazz.forBlock = creatorForItem(clazz, Inventory.TYPE.BLOCK);
+  clazz.forFilteredBlock = creatorForItem(clazz, Inventory.TYPE.FILTERED_BLOCK);
+  clazz.forMerkleBlock = creatorForItem(clazz, Inventory.TYPE.FILTERED_BLOCK);
```
Comment (throughnothing) [inline lib/messages.js]: Need to test these.

File: package.json (hunk 1 above)
Comment (maraoz) [inline package.json]: this is correct

File: lib/messages.js (hunk 2 above)
Comment (maraoz) [inline lib/messages.js]: is the type the same on purpose or a typo? so, `forMerkleBlock` and `forMerkleBlock` are aliases?
in that case I would do:
```
clazz.forMerkleBlock = clazz.forFilteredBlock;
```

Comment (throughnothing) [inline lib/messages.js]: Yeah they're the same on purpose, good call, I'll make that change.

**Comment by throughnothing** [comment]
I think this is ready again

**Comment by maraoz** [comment]
ACK. waiting for final approval by @braydonf and we'll merge this and release `bitcore-p2p`

**Comment by braydonf** [comment]
ACK

**Pull request merged**